### ISSUE TITLE
usage of sizeof(buffer) inside the snprintf call

### ISSUE THREAD

**Issue by mmd-igor** [quickfix/quickfix#575]
#https://github.com/quickfix/quickfix/commit/5978038ee405d9dfd0d7d5d78f0e3f2850784f8e
line: 6843

the potential issue lies in the usage of sizeof(buffer) inside the snprintf call. In this context, buffer is a pointer to char, so sizeof(buffer) will not give you the size of the buffer allocated for char elements but rather the size of the pointer itself, which is typically 4 or 8 bytes on most modern architectures (32-bit or 64-bit, respectively). This misunderstanding is likely causing the buffer size check to be incorrectly implemented.

**Comment by TOKocsis** [comment]
Hello!
I am a newcomer to the project, I would like to try to fix this.
Starting to work on this now.

**Comment by agmt** [comment]
Another possibility: upgrade to a new pugixml
```
curl -O --output-dir src/C++/ https://raw.githubusercontent.com/zeux/pugixml/v1.14/src/pugiconfig.hpp
curl -O --output-dir src/C++/ https://raw.githubusercontent.com/zeux/pugixml/v1.14/src/pugixml.cpp
curl -O --output-dir src/C++/ https://raw.githubusercontent.com/zeux/pugixml/v1.14/src/pugixml.hpp
```

**Comment by TOliverKocsis** [comment]
This Issue has been fixed by #587 upgrading the pugixml component by  instead of fixing this in the old version of pugixml.
So the Issue could be closed.
